spec/lib/miq_helper_spec.rb

```diff
@@ -0,0 +1,68 @@
+# This tests are done in a way that assume the Rails environment for Miq are
+# loaded, and that Rails is present to do the comparisons.  Tests that confirm
+# that this works when rails is not present us a subprocess to run the
+# `lib/miq.rb` file in isolation.
+#
+# If tests are not passing, check to see if the spec/spec_helper.rb is being
+# loaded properly and initailizing the Vmdb::Application.
+
+require 'miq_helper'
+
+describe Miq do
+  def without_rails(rb_cmd)
+    miq_lib_file = Rails.root.join("lib", "miq_helper.rb")
+    `#{Gem.ruby} -e 'require "#{miq_lib_file}"; print #{rb_cmd}'`
+  end
+
+  describe "::env" do
+    before(:each) do
```

Comment:
This does a reset of the instance variable that is cached on the class (module) level, so reseting to a clean state makes sense for each test (even though it "might" not do anything).